pedrovgs/Buzz
# Improve project design

### :tophat: What is the goal?

Improve the application look and feel by adding some icons and replacing the main screen with a custom splash screen we will evolve in the future.

### How is it being implemented?

To implement this PR I've followed these steps:

* Add icons for windows, mac and linux.
* Update the build configuration to build the app for mac and linux using ``electron-packager`` when running ``yarn dist``. You can now add ``yarn buildLinux`` or ``yarn buildMac`` if you want to build the app just for one platform.
* I've replaced the old App component with the new splash screen component we will improve in the following pull request.
* Generate an icon using a free tool and a colors template. This will be helpful in the future when reviewing the project design. Take a look at the tools I used if you think this is interesting. I've added a new README.md file into the ``art`` folder with more information related to this topic.
* I've added two new dependencies I'm going to use soon. ``electron-is-dev`` to check if the app is in development mode or not and ``material-ui`` to be able to reuse components already implemented by the material ui developers.
* Refactor the main application adding some readability.
* Configure the main application script to do not install the development tools if the application is running on production. This wasn't so easy because this script is run using raw node and I had to use an environment variable used by a different script to know if the application was running in production or not. 
* Last but not least, I changed the main window behavior to don't show the screen until the content is ready. This improves the application feedback because you don't see a white screen when the app starts.

### How can it be tested?

As the application does not have code related to the application itself I'm using the linter and the packager to check if everything goes ok. In a future PR I'll add some tests 😃
